test-data/unit/check-functools.test

```diff
@@ -0,0 +1,250 @@
+[case testTotalOrderingEqLt]
+from functools import total_ordering
+
+@total_ordering
+class Ord:
+    def __eq__(self, other: object) -> bool:
+        return False
+
+    def __lt__(self, other: "Ord") -> bool:
+        return False
+
+Ord() < Ord()
+Ord() <= Ord()
+Ord() == Ord()
+Ord() > Ord()
+Ord() >= Ord()
+
+Ord() < 1  # E: Unsupported operand types for < ("Ord" and "int")
+Ord() <= 1  # E: Unsupported operand types for <= ("Ord" and "int")
+Ord() == 1
+Ord() > 1  # E: Unsupported operand types for > ("Ord" and "int")
+Ord() >= 1  # E: Unsupported operand types for >= ("Ord" and "int")
```

Comment:
Add `reveal_type(Ord() <= Ord())` or similar to verify the type of the operation is as expected. Maybe use a non-`bool` return type in some test case.